### PR TITLE
15112424 Add README.md documentation for CircleCI configuration.

### PR DESCRIPTION
**Pull request by sleepingmonk**
Ivan asked for CI documentation with the anticipation to move to Azure at some point, and possibly other Pantheon instances in the near future.

### CODE REVIEW

File: .circleci/README.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+A CircleCI pipeline is configured via `config.yml` in the `.circleci` directory of this project.
+
+It configures a docker image that is representative of the environement we'll deploy to. In this case `quay.io/pantheon-public/build-tools-ci:6.x`. You'll want to update this if you move to a different host configuration.
+
+It sets some variables used later in the steps.
+
+It defines jobs consisting of steps. Steps are keyed with commands `run`, `save_cache`, `restore_cache`, etc.
+
+ * `run`: can define a command which executes like the commandline inside the build container. Often this simply calls scripts included in the `.circleci` directory, to do the heavy lifting.
+ * `save_cach`: define a key and a path to make accessible later in a workflow so you can avoid having to generate their contents again in the same workflow, to save some time. (i.e. `$HOME/.composer/cache`)
```
Comment (jacine) [inline .circleci/README.md]: Typo `save_cache`.

File: .circleci/README.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+* Terminus: A Pantheon commandline tool is used to interact with the pantheon account to create Multidev environments and run drush commands.
+* Multidev: (DO NOT confuse with Drupal "multisite") is what Pantheon calls extra test environments based on additional git test branches.
+
+On the creation of a Pull Request (PR) on Github, CircleCI will build the codebase from the feature branch, running composer install. When CircleCI is triggered, is configured in the settings for your project on Github.
+
+If the build completes it will create a new Pantheon Multidev environment cloning the active test database for content from `develop` branch.
+
+* The multidev environment is named after the PR id.  `pr-42`
+* You need to manually delete these multidev environments after the PR is merged to `develop` as there is a limit of 10.
```
Comment (jacine) [inline .circleci/README.md]: FYI: This depends on the hosting plan.  With the Elite plan, you can ask for more.

File: .circleci/README.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+Depending on your hosting platform, certain things may need to occur in order to deploy your new code to an environment. For example, code doesn't run directly from the git repository on the host. Code syncs to a `work_tree`.
+
+In the case of Pantheon, code is deployed to multidev environments based on branch name and standard environments based on tags.
+
+On Acquia you can set whether an environment syncs code from a branch or a tag.
+
+If you're running your own servers you can achieve code sync to a `work_tree` with a `post-recieve` git hook.
+
+After deploying a Drupal project, we usually want to run database updates, config import and cache reset. There is more than one approach you can take with this.
```
Comment (jacine) [inline .circleci/README.md]: Not usually.  Always. :)

Comment (jacine) [reply .circleci/README.md]: It’s cache “rebuild” not “reset”

File: .circleci/README.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+If your project has drush aliases configured for your hosting platform, you can add the drush commands to the deploy script or CI config file for:
+
+ * `drush @[site.env] updb -y` Database Updates
+ * `drush @[site.env] cim -y` Config Import
+ * `drush @[site.env] cr` Cache Reset
```
Comment (jacine) [inline .circleci/README.md]: Rebuild

**Review by jacine — COMMENTED** [review]
I commented on a few minor issues. Overall this is good information, but it sort of reads more like a proposal that should be in an external doc.

**Comment by jacine** [comment]
Thanks @sleepingmonk! Fixes look good, though I’m not really sure where this stands, so gonna leave it for @zakiya.

**Comment by zakiya** [comment]
I'm going to close this PR since the README has  changed a bunch anyway since it was written. JCC was tasked with clarifying this request in Jira. Still waiting for that task to come through.

**Pull request closed without merging**